lib/src/source.dart

```diff
@@ -180,21 +205,14 @@ abstract class BoundSource {
   ///
   /// This method is effectively protected: subclasses must implement it, but
   /// external code should not call this. Instead, call [getVersions].
+  @visibleForOverriding
   Future<List<PackageId>> doGetVersions(PackageRef ref);
 
   /// A cache of pubspecs described by [describe].
   final _pubspecs = <PackageId, Pubspec>{};
```

Comment:
I would love to move this to the `SystemCache` package. We don't need a separate map of these for each Source.